langchain/vectorstores/chroma.ts

```diff
@@ -0,0 +1,194 @@
+    }
+    if (vectors[0].length !== this.args.numDimensions) {
+      throw new Error(
+        `Vectors must have the same length as the number of dimensions (${this.args.numDimensions})`
+      );
+    }
+
+    const collection = await this.index.getCollection(this.collection_name);
+    for (let i = 0; i < vectors.length; i += 1) {
+      await collection.add(
+        i.toString(),
+        vectors[i]
+      )
+      this.docstore[i] = documents[i];
+    }
+  }
+
+  async similaritySearchVectorWithScore(query: number[], k: number) {
+    if (!this.index) {
+      throw new Error(
+        "Vector store not initialised yet. Try calling `addTexts` first."
+      );
+    }
+    const collection = await this.index.getCollection(this.collection_name);
+    const result = await collection.query(query, k);
+    const {ids, distances} = result;
+
+    // ids comes back as a list of lists, so we need to flatten it
+    let takeIds = ids[0]
+
+    var results = [];
+    for (let i = 0; i < takeIds.length; i += 1) {
+      takeIds[i] = parseInt(takeIds[i]);
+      results.push([this.docstore[takeIds[i]], distances[i]] as [Document, number]);
+    }
+    return results;
+  }
+
+  async save(directory: string) {
+    if (!this.index) {
+      throw new Error(
+        "Vector store not initialised yet. Try calling `addTexts` first."
+      );
+    }
+    await fs.mkdir(directory, { recursive: true });
+    await Promise.all([
+      await fs.writeFile(
+        path.join(directory, "args.json"),
+        JSON.stringify(this.args)
+      ),
+      await fs.writeFile(
+        path.join(directory, "docstore.json"),
+        JSON.stringify(this.docstore)
+      ),
+    ]);
+  }
+
+  static async load(directory: string, embeddings: Embeddings, collection_name: string) {
+    if (ChromaClient === null) {
+      throw new Error(
+        "Please install chromadb as a dependency with, e.g. `npm install -S chromadb`"
+      );
+    }
+    const args = JSON.parse(
+      await fs.readFile(path.join(directory, "args.json"), "utf8")
+    );
+
+    args.collection_name = collection_name
+
+    const index = new ChromaClient("http://localhost:8000");
+    const [docstore] = await Promise.all([
+      fs
+        .readFile(path.join(directory, "docstore.json"), "utf8")
+        .then(JSON.parse),
+    ]);
+
+    return new Chroma(args, embeddings, docstore, index);
+  }
+
+  static async fromTexts(
+    texts: string[],
+    metadatas: object[],
+    embeddings: Embeddings,
+    collection_name: string
+  ): Promise<Chroma> {
+    const docs = [];
+    for (let i = 0; i < texts.length; i += 1) {
+      const newDoc = new Document({
+        pageContent: texts[i],
+        metadata: metadatas[i],
+      });
+      docs.push(newDoc);
+    }
+    return Chroma.fromDocuments(docs, embeddings, collection_name);
+  }
+
+  static async fromDocuments(
+    docs: Document[],
+    embeddings: Embeddings,
+    collection_name: string
```

Comment:
one more, do we want to give this a default value (eg. a uuid, if you support that as a collection name, maybe prepended with `langchain` or something)? Would make it even easier to get started (at the cost of ending up with less predictable collection names in your chromadb instance)

Reply:
great idea, ill add that now